omniledger/contracts/coins.go

```diff
@@ -58,23 +59,28 @@ func (s safeUint64) sub(a uint64) (safeUint64, error) {
 // You can only delete a contractCoin instance if the account is empty.
 func ContractCoin(cdb omniledger.CollectionView, inst omniledger.Instruction, c []omniledger.Coin) (sc []omniledger.StateChange, cOut []omniledger.Coin, err error) {
 	cOut = c
+
+	var darcID darc.ID
+	_, _, darcID, err = cdb.GetValues(inst.InstanceID.Slice())
```

Comment:
I propose you get both the `value` and the `darcID` here, so you don't have to re-get it in `case InvokeType` and `case DeleteType` again.